### PR TITLE
adding view selected layers with Develop rebase

### PR DESCRIPTION
**Pull request by cphelefu**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

__Add the URL for your deploy!__

# Checklist:

- [ ] I have build __(rush build)__ and deploy __(rush host)__ my PR
- [ ] I have connected the issues(s) to this PR
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have created new issue(s) related to the outcome of this PR is needed
-  ~~I have made corresponding changes to the documentation~~
-  ~~I have added tests that prove my fix is effective or that my feature works~~
-  ~~New and existing unit tests pass locally with my changes~~

- - -
This change is [Reviewable](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1346)

### CODE REVIEW

File: packages/geoview-core/src/core/components/legend-2/legend.tsx [hunk 1]
```diff
@@ -40,6 +41,29 @@ export function Legend2(props: LegendItemsDetailsProps): JSX.Element {
...
+  const [isSelectedLayersClicked, setIsSelectedLayersClicked] = useState(false);
+
+  const handleBoxClick = () => {
+    setIsSelectedLayersClicked(!isSelectedLayersClicked);
+  };
+
+  function showSelectedLayersPanel() {
+    return (
+      <Box
+        onClick={handleBoxClick}
```
Comment (cphelefu) [inline packages/geoview-core/src/core/components/legend-2/legend.tsx:53]: onSelectedLayersClick

File: packages/geoview-core/src/core/components/legend-2/selected-layers/selected-layers-details.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
+import { Box, Grid } from '@/ui';
+
+// const sxClasses = {
```
Comment (cphelefu) [inline packages/geoview-core/src/core/components/legend-2/selected-layers/selected-layers-details.tsx:3]: TODO: Remove this unused commented code.

File: packages/geoview-core/src/core/components/legend-2/legend.tsx [hunk 3]
```diff
@@ -40,6 +41,29 @@ export function Legend2(props: LegendItemsDetailsProps): JSX.Element {
...
+        sx={{
+          display: 'flex',
+          flexDirection: 'row',
+          justifyContent: 'space-between',
+          margin: '20px',
+          cursor: 'pointer',
+        }}
+      >
+        Selected Layers
```
Comment (cphelefu) [inline packages/geoview-core/src/core/components/legend-2/legend.tsx:62]: Take this text from translations.

**Review by jolevesq — CHANGES_REQUESTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1346)* status: 0 of 8 files reviewed, 10 unresolved discussions (waiting on @cphelefu)

___
*[`packages/geoview-core/public/locales/en/translation.json` line 60 at r2](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1346#-NfzciYC344fiaTzTEpl:-NfzciYC344fiaTzTEpm:b-tpz2r2) ([raw file](https://github.com/Canadian-Geospatial-Platform/geoview/blob/cf96946ae48b7ee3f1c9fb0769755965bb263988/packages/geoview-core/public/locales/en/translation.json#L60)):*
> ```JSON
>     "selected_legend": "Selected Legend",
>     "sort_layers": "Sort Layers",
>     "selected_Layers": "An overview of currrently selected layers",
> ```

If you add item for en, you need something in the fr file
___
*[`packages/geoview-core/src/core/components/legend-2/legend.tsx` line 53 at r1](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1346#gh-1337996851:-Nfzd0Ug3jkoIVPDbMTC:bh72b12) ([raw file](https://github.com/Canadian-Geospatial-Platform/geoview/blob/9045c7b4153a852b2632c3323318548c0c63498f/packages/geoview-core/src/core/components/legend-2/legend.tsx#L53)):*
<details><summary><i>Previously, cphelefu (Christopher Phelefu) wrote…</i></summary><blockquote>

onSelectedLayersClick
</blockquote></details>

When we click on layer inside the group, the panel on the right is not updated for the selected layer.
___
*[`packages/geoview-core/src/core/components/legend-2/legend.tsx` line 37 at r2](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1346#-Nfzcrtc2uVEemQQ1x--:-Nfzcrtc2uVEemQQ1x-0:by81mi8) ([raw file](https://github.com/Canadian-Geospatial-Platform/geoview/blob/cf96946ae48b7ee3f1c9fb0769755965bb263988/packages/geoview-core/src/core/components/legend-2/legend.tsx#L37)):*
> ```XML
>     legendTitle: {
>       textAlign: 'left',
>       fontFamily: 'Open Sans, Semibold',
> ```

Should it be in the theme?
___
*[`packages/geoview-core/src/core/components/legend-2/legend-icon-list.tsx` line 4 at r2](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1346#-NfzdPLC7sI1jM_u_Sv8:-NfzdPLC7sI1jM_u_Sv9:bitupt0) ([raw file](https://github.com/Canadian-Geospatial-Platform/geoview/blob/cf96946ae48b7ee3f1c9fb0769755965bb263988/packages/geoview-core/src/core/components/legend-2/legend-icon-list.tsx#L4)):*
> ```XML
> import React, { useState, useEffect } from 'react';
> import { useTheme, Theme } from '@mui/material/styles';
> import { Checkbox, Table, TableBody, TableCell, TableContainer, TableHead, TableRow } from '@mui/material';
> ```

We should not have material ui in component, it should came from the ui, ?Do we have to use table?
___
*[`packages/geoview-core/src/core/components/legend-2/legend-icon-list.tsx` line 175 at r2](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1346#-NfzeDFY4whAd5YL6Zhp:-NfzeDFY4whAd5YL6Zhq:boo2waj) ([raw file](https://github.com/Canadian-Geospatial-Platform/geoview/blob/cf96946ae48b7ee3f1c9fb0769755965bb263988/packages/geoview-core/src/core/components/legend-2/legend-icon-list.tsx#L175)):*
> ```XML
>         <TableHead>
>           <TableRow sx={sxClasses.tableHeader}>
>             <TableCell>Name</TableCell>
> ```

This need to be in translation
___
*[`packages/geoview-core/src/core/components/legend-2/legend-item.tsx` line 319 at r2](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1346#-NfzeKKi86v1fujDq94x:-NfzeKKi86v1fujDq94y:b-e9s4ti) ([raw file](https://github.com/Canadian-Geospatial-Platform/geoview/blob/cf96946ae48b7ee3f1c9fb0769755965bb263988/packages/geoview-core/src/core/components/legend-2/legend-item.tsx#L319)):*
> ```XML
>     });
> 
>     const legendDetails = document.querySelector('#legend-details-container');
> ```

What happen if we have many maps, there is no selection by map id?
___
*[`packages/geoview-core/src/core/components/legend-2/legend-item-details/legend-item-details.tsx` line 179 at r2](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1346#-NfzeZwm-p9OBbrCCAjg:-NfzeZwm-p9OBbrCCAjh:bo2l26f) ([raw file](https://github.com/Canadian-Geospatial-Platform/geoview/blob/cf96946ae48b7ee3f1c9fb0769755965bb263988/packages/geoview-core/src/core/components/legend-2/legend-item-details/legend-item-details.tsx#L179)):*
> ```XML
>       isGroup = true;
>     }
>     console.log('is GROUP', isGroup);
> ```

Can we remove console statement
___
*[`packages/geoview-core/src/core/components/legend-2/legend-item-details/legend-item-details.tsx` line 183 at r2](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1346#-NfzedISAMPgIc5rvRdh:-NfzedISAMPgIc5rvRdi:bo2l26f) ([raw file](https://github.com/Canadian-Geospatial-Platform/geoview/blob/cf96946ae48b7ee3f1c9fb0769755965bb263988/packages/geoview-core/src/core/components/legend-2/legend-item-details/legend-item-details.tsx#L183)):*
> ```XML
>   };
> 
>   console.log(groupItems);
> ```

Can we remove console statement

**Comment by cphelefu** [comment]
*[`packages/geoview-core/src/core/components/legend-2/legend-item.tsx` line 319 at r2](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1346#-NfzeKKi86v1fujDq94x:-Ng-WG8M6cyr0VASxF4U:baf1rml) ([raw file](https://github.com/Canadian-Geospatial-Platform/geoview/blob/cf96946ae48b7ee3f1c9fb0769755965bb263988/packages/geoview-core/src/core/components/legend-2/legend-item.tsx#L319)):*
<details><summary><i>Previously, jolevesq (Johann Levesque) wrote…</i></summary><blockquote>

What happen if we have many maps, there is no selection by map id?
</blockquote></details>

Created issue to fix in another PR
https://github.com/Canadian-Geospatial-Platform/geoview/issues/1363

**Comment by cphelefu** [comment]
*[`packages/geoview-core/src/core/components/legend-2/legend-icon-list.tsx` line 4 at r2](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1346#-NfzdPLC7sI1jM_u_Sv8:-Ng-WXFs5jsgLXOIJ9iS:bp47s6p) ([raw file](https://github.com/Canadian-Geospatial-Platform/geoview/blob/cf96946ae48b7ee3f1c9fb0769755965bb263988/packages/geoview-core/src/core/components/legend-2/legend-icon-list.tsx#L4)):*
<details><summary><i>Previously, jolevesq (Johann Levesque) wrote…</i></summary><blockquote>

We should not have material ui in component, it should came from the ui, ?Do we have to use table?
</blockquote></details>

Created an issue for this issue.
https://github.com/Canadian-Geospatial-Platform/geoview/issues/1362

**Comment by cphelefu** [comment]
*[`packages/geoview-core/src/core/components/legend-2/legend.tsx` line 53 at r1](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1346#gh-1337996851:-Ng-X-xx0jEe_KWsTr8p:bhhncwc) ([raw file](https://github.com/Canadian-Geospatial-Platform/geoview/blob/9045c7b4153a852b2632c3323318548c0c63498f/packages/geoview-core/src/core/components/legend-2/legend.tsx#L53)):*
<details><summary><i>Previously, jolevesq (Johann Levesque) wrote…</i></summary><blockquote>

When we click on layer inside the group, the panel on the right is not updated for the selected layer.
</blockquote></details>

Created an issue for this.
https://github.com/Canadian-Geospatial-Platform/geoview/issues/1364

**Comment by cphelefu** [comment]
*[`packages/geoview-core/src/core/components/legend-2/legend.tsx` line 37 at r2](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1346#-Nfzcrtc2uVEemQQ1x--:-Ng-X6sB5THnffCSUtAf:b-ckmacd) ([raw file](https://github.com/Canadian-Geospatial-Platform/geoview/blob/cf96946ae48b7ee3f1c9fb0769755965bb263988/packages/geoview-core/src/core/components/legend-2/legend.tsx#L37)):*
<details><summary><i>Previously, jolevesq (Johann Levesque) wrote…</i></summary><blockquote>

Should it be in the theme?
</blockquote></details>

Leaving this here for now; will start a converation in slack to fix this in my next PR.

**Review by cphelefu — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1346)* status: 0 of 8 files reviewed, 8 unresolved discussions (waiting on @jolevesq)

___
*[`packages/geoview-core/public/locales/en/translation.json` line 60 at r2](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1346#-NfzciYC344fiaTzTEpl:-Ng-XEwVBwj5laQLo2Bx:b-896fix) ([raw file](https://github.com/Canadian-Geospatial-Platform/geoview/blob/cf96946ae48b7ee3f1c9fb0769755965bb263988/packages/geoview-core/public/locales/en/translation.json#L60)):*
<details><summary><i>Previously, jolevesq (Johann Levesque) wrote…</i></summary><blockquote>

If you add item for en, you need something in the fr file
</blockquote></details>

Done.
___
*[`packages/geoview-core/src/core/components/legend-2/legend-icon-list.tsx` line 175 at r2](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1346#-NfzeDFY4whAd5YL6Zhp:-Ng-WP5KC-cpOVzY9vcj:b-896fix) ([raw file](https://github.com/Canadian-Geospatial-Platform/geoview/blob/cf96946ae48b7ee3f1c9fb0769755965bb263988/packages/geoview-core/src/core/components/legend-2/legend-icon-list.tsx#L175)):*
<details><summary><i>Previously, jolevesq (Johann Levesque) wrote…</i></summary><blockquote>

This need to be in translation
</blockquote></details>

Done.
___
*[`packages/geoview-core/src/core/components/legend-2/legend-item-details/legend-item-details.tsx` line 179 at r2](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1346#-NfzeZwm-p9OBbrCCAjg:-Ng-VJTW54IaMfTHEoov:b-896fix) ([raw file](https://github.com/Canadian-Geospatial-Platform/geoview/blob/cf96946ae48b7ee3f1c9fb0769755965bb263988/packages/geoview-core/src/core/components/legend-2/legend-item-details/legend-item-details.tsx#L179)):*
<details><summary><i>Previously, jolevesq (Johann Levesque) wrote…</i></summary><blockquote>

Can we remove console statement
</blockquote></details>

Done.
___
*[`packages/geoview-core/src/core/components/legend-2/legend-item-details/legend-item-details.tsx` line 183 at r2](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1346#-NfzedISAMPgIc5rvRdh:-Ng-VJx2D3_SaJIZ1lr4:b-896fix) ([raw file](https://github.com/Canadian-Geospatial-Platform/geoview/blob/cf96946ae48b7ee3f1c9fb0769755965bb263988/packages/geoview-core/src/core/components/legend-2/legend-item-details/legend-item-details.tsx#L183)):*
<details><summary><i>Previously, jolevesq (Johann Levesque) wrote…</i></summary><blockquote>

Can we remove console statement
</blockquote></details>

Done.

**Pull request merged**